### ISSUE TITLE
[INVESTIGATION] Continuous Integration Tests Not Reliably Passing

### ISSUE THREAD

**Issue by shea-hawkins** [scalanlp/breeze#652]
Hello,

I took a look at the build logs to see if I could help with this issue & fix a test, but looks like it's a tricky one because it's a different test failing each time. I've started with collecting some information about the failing tests, beginning with the last three master builds, in case there's a randomized component that is shared between them all.

I'd like to investigate further if you can point me in the right direction:
**BUILD 1039.1**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.10.6 test_
- stupidly simple mcmc *** FAILED *** (693 milliseconds)
[info]   0.05023239184840911 was not less than 0.05 (metropolisTest.scala:52)

**BUILD 1039.2**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.11.8 test_
All tests passed.

**BUILD 1039.3**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.12.1 test_
[info] - exp(digamma(x)) ≈ x - .5, x >= 10 *** FAILED *** (24 milliseconds)
[info]   GeneratorDrivenPropertyCheckFailedException was thrown during property evaluation.
[info]    (NumericsTest.scala:85)
[info]     Falsified after 8 successful property evaluations.
[info]     Location: (NumericsTest.scala:85)
[info]     Occurred when passed generated values (
[info]       arg0 = -12.363934295188095 // 2 shrinks
[info]     )
[info]   org.scalatest.exceptions.GeneratorDrivenPropertyCheckFailedException:

**BUILD 1032.1**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.10.6 test_
[info] dot product associates with scalar multiplication *** FAILED *** (28 milliseconds)
[info]   GeneratorDrivenPropertyCheckFailedException was thrown during property evaluation.
[info]    (TensorSpaceTestBase.scala:64)
[info]     Falsified after 4 successful property evaluations.
[info]     Location: (TensorSpaceTestBase.scala:64)
[info]     Occurred when passed generated values (
[info]       arg0 = (SparseVector(9)((2,1.0E-4), (4,50304.0), (5,1.0000000000000007E-4)),SparseVector(9)((0,-31936.0), (2,-1.0E-4), (5,1.0E-4), (6,-1.0E-4), (7,-58272.0), (8,80290.078125)),SparseVector(9)()),
[info]       arg1 = 17856.0
[info]     )

**BUILD 1032.2**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.12.1 test_
- stupidly simple mcmc *** FAILED *** (635 milliseconds)
[info]   0.05061354264990836 was not less than 0.05 (metropolisTest.scala:52)
[info]   org.scalatest.exceptions.TestFailedException:

**BUILD 1032.3**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.12.1 test_
[info] - stupidly simple mcmc *** FAILED *** (635 milliseconds)
[info]   0.05061354264990836 was not less than 0.05 (metropolisTest.scala:52)
[info]   org.scalatest.exceptions.TestFailedException:

**BUILD 1030.1**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.10.6 test_
All tests passed.

**BUILD 1030.2**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.11.8 test_
- stupidly simple mcmc *** FAILED *** (558 milliseconds)
[info]   0.052514521173608664 was not less than 0.05 (metropolisTest.scala:52)
[info]   org.scalatest.exceptions.TestFailedException:

**BUILD 1030.3**
_sbt -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.F2jBLAS -Dcom.github.fommil.netlib.LAPACK=com.github.fommil.netlib.F2jLAPACK -Dcom.github.fommil.netlib.ARPACK=com.github.fommil.netlib.F2jARPACK ++2.12.1 test_
[info] - stupidly simple mcmc *** FAILED *** (552 milliseconds)
[info]   0.056047782992191486 was not less than 0.05 (metropolisTest.scala:52)
[info]   org.scalatest.exceptions.TestFailedException:

**Comment by dlwh** [comment]
Hi Shea, sorry for not responding. Yeah, we use unseeded RNGs and the
sensitivities are all set poorly. I used to have them under control, but I
don't anymore. The MCMC ones should be easy to fix. The other ones are
conditional on Scalacheck and I can't figure out how to control its RNG

**Comment by superbobry** [comment]
Well, it seems ~6 months later Scalacheck still does not have a release allowing to seed the RNG. See rickynils/scalacheck#374.

**Comment by dlwh** [comment]
yeah. the builds have gotten much more stable because I've tweaked
tolerances and such (the 2.10.6 build deterministically failing
notwithstanding), but it would be great to have that
